### PR TITLE
tests: clean up after NFS tests

### PR DESCRIPTION
**Pull request by zyga**
The test was leaking two mount points: /var/lib/nfs/rpc_pipefs and
/proc/fs/nfsd. In addition because of how the test operates, files
placed in the current working directory can get "lost" somehow. As a
precaution I moved all of the helper files to /tmp.

Note that the entire test is disabled on Fedora, CentOS, openSUSE and
Ubuntu Core so there is still some "dead code" in terms of switch
statements that are not executed. I tried to do my best to keep the dead
code as useful, to the future developers, as I could.

There are small unrelated changes as well, like how /home is unmounted
or a FIXME note about test clobbering a file on Fedora.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

File: tests/main/nfs-support/task.yaml [hunk 1]
```diff
@@ -41,6 +54,46 @@ restore: |
     systemctl reset-failed snapd.service
     systemctl start snapd.service
 
+    # Depending on OS do cleanup appropriate for the host.
+    case "$SPREAD_SYSTEM" in
+        ubuntu-14.04-*)
```
Comment (mvo5) [inline tests/main/nfs-support/task.yaml]: Why is ubuntu-16.04, 18.04 etc missing here?

Comment (mvo5) [reply tests/main/nfs-support/task.yaml]: Maybe a comment here for our future selfs that may have the same question?

Comment (zyga) [reply tests/main/nfs-support/task.yaml]: I'll add a comment but the basic answer is that the cleanup is split between two chunks (stop, unmount, start-if-needed). Not all systems need the changes, specifically Ubuntu 16.04 and beyond have NFS preinstalled and working and we don't need to stop it.

Comment (zyga) [reply tests/main/nfs-support/task.yaml]: Can you look again?

File: tests/main/nfs-support/task.yaml [hunk 2]
```diff
@@ -75,8 +128,9 @@ execute: |
             systemctl restart nfs-kernel-server
             ;;
         fedora-*)
-            # Enable udp protocol for nfs on fedora which is disable by default           
+            # Enable udp protocol for nfs on fedora which is disable by default
```
Comment (bboozzoo) [inline tests/main/nfs-support/task.yaml]:
```suggestion
            # Enable udp protocol for nfs on fedora which is disabled by default
```

**Pull request merged**